使用 RepeatMasker（version 4.1.1）和 RepeatMolder（version 2.0.2）两款主流软件对基因组重复序列进行从头预测。
首先使用 RepeatMolder 通过 BLAST的方法对甘蔗高贵种基因组序列自身之间进行比对，选择高重复的序列来构建本物种特有的重复序列数据库。
然后使用 Repeatmasker 加载构建好的数据库并进行全面的注释，鉴定得到基因组中的重复序列。
对于未知的重复序列，使用软件TEclass（version 2.1.3）.对其重注释。最后将基因组中的重复序列进行屏蔽，
方便于后续的基因模型预测。

##（1）RepeatModeler 预测基因组重复序列，为下一步的RepeatMasker建立重复序列数据库

```
#建库，利用rice 一个实例fasta文件

BuildDatablase –name ricedb –engine ncbi  rice.sample.2.fa

#参数说明

-name <database name>
        The name of the database to create.
-engine <engine name>
        The name of the search engine we are using. I.e abblast/wublast or 
-dir <directory>
        The name of a directory containing fasta files to be processed. The
        files are recognized by their suffix. Only *.fa and *.fasta files
        are processed.
-batch <file>
        The name of a file which contains the names of fasta files to
        process. The files names are listed one per line and should be full
        qualified.
RepeatModeler -database ricedb  -pa 5
        
#参数说明
        
-database
        The name of the sequence database to run an analysis on. This is th
        name that was provided to the BuildDatabase script using the "-name
        option.
-pa 
        Specify the number of parallel search jobs to run. RMBlast jobs wil
        use 4 cores each and ABBlast jobs will use a single core each. i.e.
        on a machine with 12 cores and running with RMBlast you would use
        -pa 3 to fully utilize the machine.
-recoverDir <Previous Output Directory>
        If a run fails in the middle of processing, it may be possible
        recover some results and continue where the previous run left off.
        Simply supply the output directory where the results of the failed
        run were saved and the program will attempt to recover and continue
        the run.
```
 
##（1.1）RepeatModeler结果文件

consensi.fa：自身比对找到的一致性序列
consensi.fa.classified：重复序列分类结果，“#”后面的字符串是分类结果
consensi.fa.masked：屏蔽后的序列，不太使用

##（2）RepeatMasker；屏蔽DNA序列中转座子或者低复杂度重复序列，输出的序列把已知的重复序列都屏蔽成N或X，并给出相应的位置和统计信息
        
```
#将上一步的结果文件（自动生成，以“RM”开头）里面的consensi.fa.classified为libirary，对参考序列进行序列屏蔽
#运行
RepeatMasker -nolow  -e ncbi -pa 5 -norna -dir ./  -lib consensi.fa.classified rice.sample.fa
#重要参数
-nolow. (不屏蔽低复杂序列，担心屏蔽后使得基因结构受到影响)
        Does not mask low_complexity DNA or simple repeats
-lib [filename] 自己制定的重复序列库
        Allows use of a custom library (e.g. from another species)
 -e(ngine) [crossmatch|wublast|abblast|ncbi|rmblast|hmmer]
        Use an alternate search engine to the default.
-pa(rallel) [number]
        The number of sequence batch jobs [50kb minimum] to run in parallel.
-norna
        Does not mask small RNA (pseudo) genes
-dir [directory name] 输出文件位置
        Writes output to this directory (default is query file directory,
        "-dir ." will write to current directory).
-gff
        Creates an additional Gene Feature Finding format output
```
        
        
